### PR TITLE
Fixes #1428: support "not a number" values for 'float', 'double' columns

### LINKED ISSUE #1428: Tables number codec: support "NaN", "Infinity

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Add support, tests, for NaN/FP values.

**Which issue(s) this PR fixes**:
Fixes #1428

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/InsertOneTableIntegrationTest.java [hunk 1]
```diff
@@ -48,6 +49,18 @@ public final void createDefaultTables() {
             "bigIntegerValue",
             Map.of("type", "varint")),
         "id");
+    createTableWithColumns(
```
Comment (Yuqi-Du) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/InsertOneTableIntegrationTest.java:52]: So for Integration test, we still need to create the table with correct columns to test against, correct?
The previous thing Aaron showed for all kinds of code combination, that is unit test?

**Review by Yuqi-Du — APPROVED** [review]
LGTM!

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/InsertOneTableIntegrationTest.java (hunk 1 above)
Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/tables/InsertOneTableIntegrationTest.java:52]: Correct on both: Aaron has been writing unit tests, with more scaffolding. I added some unit tests too, but those are simpler parameterized codec tests.

ITs do use some new helper classes but do require old-style table creation.
We will probably need a way to create tables with direct CQL but for now use of `createTable` works.

**Review by vkarpov15 — COMMENTED** [review]
LGTM. I can't say the approach of sending NaN, Infinity, -Infinity as strings is my favorite, but I can't think of a better approach and it is reasonable given that we know to expect a number on the data api side.

**Comment by tatu-at-datastax** [comment]
@vkarpov15 yes, it's not ideal but will do for now. And can be still revisited before GA based on feedback. In theory we could use EJSON wrapper, fwtw, but even there I think it'd be Number-or-String as wrapped value.

**Pull request merged**